app/src/main/report/components/ReportsList/ReportsList.ts

```diff
@@ -0,0 +1,41 @@
+import { connect } from 'react-redux';
+import { Dispatch } from "redux";
+import { compose, lifecycle, branch, renderComponent } from "recompose";
+import { createSelector } from "reselect";
+
+import {ReportAppState} from "../../reducers/reportAppReducers";
+import { reportsActions } from "../../actions/reportsActions";
+import {ReportsListComponent, ReportsListComponentProps} from "./ReportsListComponent";
+import { LoadingElement } from "../../../shared/partials/LoadingElement/LoadingElement";
+import {getDisplayedReportsListSelector} from "./reportsListSelectors";
+
+export interface ReportsListContainerProps extends ReportsListComponentProps {
+    getReports: () => void;
+    ready: boolean;
+}
+
+const lifecyleProps = {
+    componentDidMount() {
+        this.props.getReports();
```

Comment:
I'm a little confused here - this logic makes less sense when decoupled from a component. What does `this` refer to, and how do we know that `this.props` exists or has a property called `getReports`? We're just relying on this object being mapped correctly into a component of the right type to hook up the `this` keyword in an appropriate way; is there a way with recompose to enforce this a bit more rigorously?